operator/src/main/java/oracle/kubernetes/operator/Namespaces.java

```diff
@@ -121,15 +125,40 @@ private String getConfiguredNamespaceList() {
       }
 
       @Override
-      public boolean isDomainNamespace(@Nonnull String namespaceName) {
-        return true;  // filtering is done by Kubernetes list call
+      public boolean isDomainNamespace(@Nonnull V1ObjectMeta nsMetadata) {
+        // although filtering is done by Kubernetes list call, there is a rice condition where readExistingNamespaces
```

Comment:
I'm trying to imagine what a "rice" condition could be. :)

Can you explain what this race condition is? Why would K8s give us a non-matching namespace?

Reply:
The intermittent issue only happens when the selection strategy is changed from List to LabelSelector. When the readExistingNamespaceAysnc is called, the strategy may still be List, so the K8S returns all namespaces.